src/org/rust/lang/annotator/RustAnnotator.kt

```diff
@@ -29,6 +28,26 @@ public class RustAnnotator : Annotator {
             is RustTypeParam -> {
                 addTextAttributes(element, holder, RustColors.TYPE_PARAMETER)
             }
+            is RustPatIdent -> {
+                if (element.bindingMode?.mut != null) {
+                    addTextAttributes(element.identifier, holder, RustColors.MUT_BINDING)
+                }
+            }
+            is RustPathPart -> {
+                val resolved = RustResolveEngine().resolve(element).resolved
+                when (resolved) {
+                    is RustPatField -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                    is RustPatIdent -> {
+                        if (resolved.bindingMode?.mut != null) {
+                            addTextAttributes(element, holder, RustColors.MUT_BINDING)
+                        }
+                    }
+                }
```

Comment:
We will get a better PSI structure, reflected in the grammar. Also, this will fix a bug in the grammar :)

Consider the rule for `pat_ident`:

`pat_ident       ::= binding_mode IDENTIFIER [AT pat]`

With, and without `@ pat` this rule means completely different things. Without pat, it binds IDENTIFIER to some while, but with pat it does not bind identifier and instead destructs a struct field. 

Also, current  grammar incorrectly accepts the following:

```Rust
struct S {
    f: f64
}

fn main() {
    let S {mut f : f} = S { f: 92.0 };
}
```